packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart

```diff
@@ -442,27 +442,48 @@ class _ExperimentalResidentWebRunner extends ResidentWebRunner {
       progressId: 'hot.restart',
     );
 
-    final UpdateFSReport report = await _updateDevFS(fullRestart: fullRestart);
-    if (report.success) {
-      device.generator.accept();
+    String reloadModules;
+    if (!debuggingOptions.buildInfo.isDebug) {
+      try {
+        await buildWeb(
+          flutterProject,
+          target,
+          debuggingOptions.buildInfo,
+          debuggingOptions.initializePlatform,
+          dartDefines,
+          false,
+        );
+      } on ToolExit {
+        return OperationResult(1, 'Failed to recompile application.');
```

Comment:
Yes, it will be printed as an error before the exit:

```
Launching lib/main.dart on Chrome in release mode...
Target dart2js failed: Exception: lib/main.dart:7:99:                   
Error: Expected ';' after this.
void main() => runApp(const Center(child: Text('Hello, world!', textDirection: TextDirection.ltr)))
                                                                                                  ^
lib/main.dart:8:1:
Error: Unexpected token ''.

Error: Compilation failed.


Compiling lib/main.dart for the Web...                              4.2s
Exception: Failed to compile application for the Web.

```